lib/spack/spack/util/path.py

```diff
@@ -60,6 +61,62 @@
 SPACK_PATH_PADDING_CHARS = '__spack_path_placeholder__'
 
 
+def is_path_url(path):
+    url_tuple = urlparse(path)
+    if not url_tuple.netloc:
+        return False
+    return True
```

Comment:
We also have URLs that are local files, for example:

```
>>> from six.moves.urllib.parse import urlparse
>>> x = urlparse('file:///path/to/a/thing')
>>> x
ParseResult(scheme='file', netloc='', path='/path/to/a/thing', params='', query='', fragment='')
```

so I think we need to check the `scheme` as well.